tests/test_timefuncs.py

```diff
@@ -1035,3 +1036,18 @@ def test_timedelta_range() -> None:
 def test_dateoffset_freqstr() -> None:
     offset = DateOffset(minutes=10)
     check(assert_type(offset.freqstr, str), str)
+
+
+def timedelta64_and_arithmatic_operator() -> None:
+    s1 = pd.Series(data=pd.date_range("1/1/2020", "2/1/2020"))
+    s2 = pd.Series(data=pd.date_range("1/1/2021", "2/1/2021"))
+    s3 = s2 - s1
+    td = np.timedelta64(1, "M")
+    assert_type((s1 - td), TimestampSeries)
+    assert_type((s1 + td), TimestampSeries)
+    assert_type((s1 * td), NoReturn)  # pyright: ignore
+    assert_type((s1 / td), NoReturn)  # pyright: ignore
+    assert_type((s3 - td), TimedeltaSeries)
+    assert_type((s3 + td), TimedeltaSeries)
+    assert_type((s3 * td), NoReturn)  # pyright: ignore
+    assert_type((s3 / td), pd.Series[float])
```

Comment:
use `check` on this one too

Reply:
means on all three of these except the NoReturn one